raster/r.colors.out/prt_json.c

```diff
@@ -0,0 +1,252 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include <grass/gis.h>
+#include <grass/raster.h>
+#include <grass/glocale.h>
+#include <grass/parson.h>
+
+#include "local_proto.h"
+
+#define RGB_STRING_LENGTH 20
+#define HEX_STRING_LENGTH 8
+#define HSV_STRING_LENGTH 30
+
+/*!
+   \brief Converts RGB color values to HSV format.
+
+   \param r red component of the RGB color
+   \param g green component of the RGB color
+   \param b blue component of the RGB color
+   \param h pointer to store the calculated hue
+   \param s pointer to store the calculated saturation
+   \param v pointer to store the calculated value
+ */
+void rgb_to_hsv(int r, int g, int b, float *h, float *s, float *v)
+{
+    float r_norm = r / 255.0f;
+    float g_norm = g / 255.0f;
+    float b_norm = b / 255.0f;
```

Comment:
To suppress clang-tidy's [narrowing conversions](https://clang.llvm.org/extra//clang-tidy/checks/cppcoreguidelines/narrowing-conversions.html) warnings.

```suggestion
    float r_norm = (float)r / 255.0f;
    float g_norm = (float)g / 255.0f;
    float b_norm = (float)b / 255.0f;
```